src/server/pfs/pfs.go

```diff
@@ -266,11 +283,11 @@ func (e ErrBaseCommitNotFinished) Error() string {
 }
 
 func (e ErrAmbiguousCommit) Error() string {
-	return fmt.Sprintf("commit %v is ambiguous (specify the branch to resolve)", e.Commit)
+	return fmt.Sprintf("commit %v is ambiguous. Possible resolutions are: %v", e.Commit, e.PossibleCommits)
```

Comment:
Errors should compose; this should probably use a subclause instead: `"commit %v is ambiguous; possible resolutions are: %v"`.

Even better would be to join `PossibleCommits` with a function which knows how to handle commas and ‘and.’